Softeq.XToolkit.PushNotifications.iOS/Services/IosPushNotificationService.cs

```diff
@@ -0,0 +1,91 @@
+using Softeq.XToolkit.PushNotifications.iOS.Abstract;
+using UIKit;
+using UserNotifications;
+
+namespace Softeq.XToolkit.PushNotifications.iOS.Services
+{
+    /// <summary>
+    ///     Default implementation of <see cref="IPushNotificationsService"/> interface for iOS platform.
+    /// </summary>
+    public sealed class IosPushNotificationService : IPushNotificationsService, IPushNotificationAppDelegate
+    {
+        private readonly IPushNotificationsConsumer _pushNotificationsConsumer;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="IosPushNotificationService"/> class.
+        /// </summary>
+        /// <param name="pushNotificationsConsumer">Consumer of the push notification related callbacks.</param>
+        /// <param name="logManager">Provides logging.</param>
+        public IosPushNotificationService(
+            IPushNotificationsConsumer pushNotificationsConsumer,
+            ILogManager logManager)
+        {
+            _pushNotificationsConsumer = pushNotificationsConsumer;
+            _logger = logManager.GetLogger<IosPushNotificationService>();
+
+            UNUserNotificationCenter.Current.Delegate = new UserNotificationCenterDelegate(pushNotificationsConsumer, logManager);
+            UNUserNotificationCenter.Current.SetNotificationCategories(new NSSet<UNNotificationCategory>(pushNotificationsConsumer.GetCategories().ToArray()));
+        }
+
+        /// <inheritdoc />
+        public async Task RegisterForPushNotificationsAsync()
+        {
+            var requiredAuthorizationOptions = _pushNotificationsConsumer.GetRequiredAuthorizationOptions();
+            var (isGranted, _) = await UNUserNotificationCenter.Current.RequestAuthorizationAsync(requiredAuthorizationOptions);
+            _pushNotificationsConsumer.OnPushNotificationAuthorizationResult(isGranted);
+
+            UIApplication.SharedApplication.InvokeOnMainThread(() =>
+            {
+                UIApplication.SharedApplication.RegisterForRemoteNotifications();
+            });
+        }
+
+        /// <inheritdoc />
+        public async Task UnregisterForPushNotificationsAsync()
+        {
+            UIApplication.SharedApplication.InvokeOnMainThread(() =>
+            {
+                UIApplication.SharedApplication.UnregisterForRemoteNotifications();
+            });
+
+            await _pushNotificationsConsumer.OnUnregisterFromPushNotifications();
+        }
+
+        /// <inheritdoc />
+        public void RegisteredForRemoteNotifications(UIApplication application, NSData deviceToken)
+        {
+            _pushNotificationsConsumer.OnRegisteredForRemoteNotifications(application, deviceToken);
+        }
+
+        /// <inheritdoc />
+        public void FailedToRegisterForRemoteNotifications(UIApplication application, NSError error)
+        {
+            _pushNotificationsConsumer.OnFailedToRegisterForRemoteNotifications(application, error);
```

Comment:
The only common method or consumers already has the exact same name - `OnUnregisterFromPushNotifications()`, I doubt any other methods can be "synced".
Regarding the naming, I tried to save the same names (and signatures) as the platform API has - my idea was that it would be simpler to understand for people who know platform APIs. But if you have suggestions for names - please provide them